neqo-qpack/src/qlog.rs

```diff
@@ -0,0 +1,66 @@
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// Functions that handle capturing QLOG traces.
+
+use neqo_common::NeqoQlogRef;
+use qlog::{
+    EventCategory, EventData::QpackInstructionReceived, EventField,
+    QPackInstruction::InsertCountIncrementInstruction, QpackInstructionTypeName,
+};
+use std::fmt::LowerHex;
+use std::time::Instant;
+
+// TODO(hawkinsw@obs.cr): There is a copy of this in neqo-transports/src/qlog.rs.
+// Refactor both uses into something in neqo-common.
+fn slice_to_hex_string<T: LowerHex>(slice: &[T]) -> String {
```

Comment:
Shouldn't the qlog crate take slices rather than strings for binary data?

If it doesn't can we provide them with a PR?

Reply:
This is just how the qlog API works presently. It got better from 0.1 -> 0.2 and I think it will continue to get better. For now, the API wants owned params, so whether we clone them or the lib does may not matter much.